src/render/idyll-display/components/component-editor/property.js

```diff
@@ -22,6 +23,15 @@ export const WrappedComponent = withContext(
       };
     }
 
+    componentDidUpdate() {
+      const { context } = this.props.context;
+      if (equal(context.data(), this.state.variableData)) {
```

Comment:
I think just comparing the keys here instead of the entire object will be faster in the case of large data sets. Would something like `equal(Object.keys(context.data()), Object.keys(this.state.variableData))` works?

Reply:
Ah that's a really good point. Will update